pkg/authenticator/config/config.go

```diff
@@ -35,11 +35,11 @@ const (
 	DefaultConjurVersion = "5"
 
 	// DefaultTokenRefreshTimeout is the default time the system waits to reauthenticate on error
-	DefaultTokenRefreshTimeout = 6 * time.Minute
+	DefaultTokenRefreshTimeout = "6m0s"
 
 	// DefaultClientCertRetryCountLimit is the amount of times we wait after successful
 	// login for the client certificate file to exist, where each time we wait for a second.
-	DefaultClientCertRetryCountLimit = 10
+	DefaultClientCertRetryCountLimit = "10"
```

Comment:
are there minimum or maximum values for this limit? Can they put `-5` for example? Would the logic hold up against negative numbers?

Reply:
In SP we added a check for `envValueInt < minValue` with a minValue being 0 in order to avoid this

Reply:
hmmmm. do you think it is needed or is it too low level?

Reply:
do we want to not let users set (for example) `CONJUR_TOKEN_TIMEOUT=-5`?

Reply:
I can totally see this being an issue _especially_ because it is configurable no? It is coming from an environment and I don't trust user input 🙃

Reply:
yes, but it adds another param that the function consumer needs to set where for most of the times it will be `0` and not interesting. i'll let @izgeri give her take on this before going forward.

Reply:
I think we should have error checking to ensure sane and acceptable values are input for this param. is the param currently documented? in the README or just in the CONTRIB? if it's just for dev purposes / dev troubleshooting, we could just leave undocumented for now. if we're adding it AND documenting it, then we need to be more careful (and this would need a changelog entry)

Reply:
I don't want to hold up this PR, but we should really consider doing a followup to allow for minimum/maximum range validation. In most cases, I'd bet that we can define hard-coded values for min and max values that are sensible and are liberal enough to cover all cases. For example, for certificate retries, we could set the minimum to 0 and the maximum to something astronomical (`2**32 - 1`?) If we don't add range checks, I don't think we know what would happen if someone sets retries or some other configured numbered value to -1 (I'm sure it's not in our integration testing).

Reply:
by adding a minimum value we raise a few questions:
1. what is the minimum value of `CONJUR_TOKEN_TIMEOUT`?
2. what is the minimum value of `CONJUR_CLIENT_CERT_RETRY_COUNT_LIMIT`?
3. what do we do in case the given value is less than the minimum? Do we fail the app or use the default value?
4. does the answer to question #3 changes how we handle cases where the user gave an invalid value for `CONJUR_TOKEN_TIMEOUT`?

while these are good questions, i don't want to ask them in this PR and in this release that we are preparing. Once this is merged and the release is out there, we can look back at these questions and decide properly.

Reply:
created an issue for this: https://github.com/cyberark/conjur-authn-k8s-client/issues/162